vulnerable_people_form/form_pages/shared/validation.py

```diff
@@ -295,6 +296,28 @@ def validate_phone_number_if_present(section_key, phone_number_key):
     return True
 
 
+def validate_sms_phone_number_if_present(section_key, phone_number_key):
+    phone_number = form_answers()["contact_details"].get(phone_number_key, "")
+    if phone_number and phone_number_is_valid_for_notify(phone_number):
+        return True
+    else:
+        error_message = "Enter a UK mobile number, like 07700 900000 or +44 7700 900000"
+        error_section = session.setdefault("error_items", {}).get(section_key, {})
+        session["error_items"] = {
+            **session.setdefault("error_items", {}),
+            section_key: {**error_section, phone_number_key: error_message},
+        }
+        return False
+
+
+def phone_number_is_valid_for_notify(phone_number):
+    try:
+        validate_uk_phone_number(phone_number)
+        return True
+    except InvalidPhoneError:
+        return False
```

Comment:
do we want to log at this point saying it was invalid and possibly the reason ?

Reply:
sure thing

Reply:
actually I've just reviewed our logging code and logging validation errors would be something new - fine to do it this one time only if you want but it might be better as part of a larger "extend logging to cover validation" story

Reply:
i was thinking more like a warning e.g.

`logger.warning(create_log_message(log_event_names["NHS_LOGIN_USER_CONSENT_NOT_GIVEN"],
                                              f"NHS login warning: {error}, error description: {error_description}"))`

Reply:
Okay - I've done that - the message doesn't include the phone number - didn't know whether it should or shouldn't on account of the PII factor

Reply:
thats fine i think its right not to store anything in splunk of PII